### DISCUSSION TITLE
Permettre aux utilisateurs de configurer eux meme leurs fonds de plan

### DISCUSSION THREAD

**Discussion by submarcos** [GeotrekCE/Geotrek-admin#2965] [Ideas]
Le problème et la question reviennent souvent, notamment à cause des expirations régulières des clés IGN, et des problèmes et changement fréquents de leur côté.
Il serait interessant de permettre aux utilisateurs de configurer eux meme leurs fonds de plan.

- Pour Geotrek-admin, via django-leaflet (tout en gardant le système actuel fonctionnel)

- Pour rando v3, depuis Geotrek-admin, avec une configuration par portail

- Pour rando v2, est-ce que çà vaut le coup car il est en fin de vie est plus destiné à recevoir des mises à jour, des développements sur cette plateforme me semble évitables

**Comment by mviadere-openig** [comment]
Bonjour, de mon côté je suis carrément preneur, j'aimerai d'ailleurs savoir s'il est possible de mettre en place un flux avec authentification, cela nous permettrait d'utiliser les flux carto de nos serveurs.

**Comment by submarcos** [comment]
@mviadere-openig  c'est possible mais il faut utiliser le meme systeme de clé que l'IGN, car elles doivent etre diffusées en clair dans l'url (les tuiles étant chargées directement par le navigateur, impossible d'y ajouter des en-tetes avec des jetons d'accès). Il faut donc un serveur de tuile, ou une application qui ferait serveur mandataire (proxy) qui vérifierait que la clé est utilisée avec le bon "referer" (url de la page web utilisant les tuiles)

**Comment by camillemonchicourt** [comment]
J'ai un doute.
C'est pour que les utilisateurs les configure eux-mêmes ou les administrateurs ?
Le fonctionnement actuel me convient bien car j'ai accès aux fichiers de mon serveur.

Mais si le sujet est pour les administrateurs qui n'ont pas accès à leur serveur, alors cela repose la question plus large de pouvoir accéder aux paramètres dans l'adminsite.

**Comment by submarcos** [comment]
@camillemonchicourt Oui je voulais bien sur parler des administrateurs.
çà arrive souvent, et notamment du besoin de connaitre la syntaxe, meme simpliste, de python pour réaliser cela.

**Comment by camillemonchicourt** [comment]
OK, en effet dans ce cas, je pense que cela serait intéressant de penser cela en lien avec la possibilité de gérer, modifier les paramètres de Geotrek-admin dans l'AdminSite.
